queries.sql

```diff
@@ -116,3 +116,72 @@ GROUP BY full_name
 ORDER BY animal_count DESC
 LIMIT 1;
 
+-- Who was the last animal seen by William Tatcher?
+SELECT animals.name, visits.date_of_visit FROM animals
+INNER JOIN visits ON animals.id = visits.animals_id
+WHERE visits.vets_id = (SELECT id FROM vets WHERE name = 'William Tatcher')
+ORDER BY visits.date_of_visit DESC
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT count(*) FROM animals
+INNER JOIN visits ON animals.id = visits.animals_id
+WHERE visits.vets_id = (SELECT id FROM vets WHERE name = 'Stephanie Mendez');
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT vets.name, (SELECT name as specialization FROM species WHERE specializations.species_id = species.id) FROM vets
+LEFT JOIN specializations ON vets.id = specializations.vets_id;
+
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT animals.name FROM animals
+LEFT JOIN visits ON animals.id = visits.animals_id
+WHERE visits.date_of_visit BETWEEN 'April 1, 2020' AND 'August 30, 2020';
+
+-- What animal has the most visits to vets?
+SELECT animals.name, count(visits.date_of_visit) AS visit_count  FROM animals
+LEFT JOIN visits ON animals.id = visits.animals_id
+GROUP BY animals.name
+ORDER BY visit_count DESC
+LIMIT 1;
+
+-- Who was Maisy Smith's first visit?
+SELECT animals.name FROM animals
+LEFT JOIN visits ON animals.id = visits.animals_id
+ORDER BY visits.date_of_visit
+LIMIT 1;
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT 
+    animals.name,
+    animals.date_of_birth,
+    animals.escape_attempts,
+    animals.neutered,
+    animals.weight_kg,
+    vets.name,
+    vets.age,
+    vets.date_of_graduation 
```

Comment:
[OPTIONAL]

- Here you can use like `animals.*`, `vets.*` to select all the columns of the table.